buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/RandomizedTestingTask.groovy

```diff
@@ -95,6 +97,17 @@ class RandomizedTestingTask extends DefaultTask {
         listenersConfig.listeners.add(new TestReportLogger(logger: logger, config: testLoggingConfig))
     }
 
+    void basedOn(RandomizedTestingTask other) {
+        configure(BuildPlugin.commonTestConfig(project))
+        classpath = other.classpath;
+        testClassesDirs = other.testClassesDirs;
```

Comment:
I think all the users of this could just be converted to use this line? I don't think we need to or should overly abstract this "shared code". For example, it would be better to make internalClusterTest task added by an additional gradle plugin to reduce duplication there.

Reply:
I did change just the line everywhere during development, and decided to take the extra step to refactor it as I felt it's too detailed and does not convey intent well enough, every time I encountered it I had to carefully check to make sure none of the configuration is missing or slightly different. I do agree that a plugin as you suggest would be even better,  but this is a step that will make that a bit easier to implement.